ngraph/core/src/op/detection_output.cpp

```diff
@@ -45,16 +45,227 @@ op::DetectionOutput::DetectionOutput(const Output<Node>& box_logits,
+                        std::to_string(prior_box_size) + "). Current value is: ",
+                    proposals_pshape[2].get_length(),
+                    ".");
+                num_prior_boxes = proposals_pshape[2].get_length() / prior_box_size;
+            }
+            else
+            {
+                int num_prior_boxes_val = num_prior_boxes.get_length();
+                NODE_VALIDATION_CHECK(this,
+                                      proposals_pshape[2].get_length() ==
+                                          num_prior_boxes_val * prior_box_size,
+                                      "Proposals' third dimension must be equal to num_prior_boxes "
+                                      "* prior_box_size (" +
+                                          std::to_string(num_prior_boxes_val * prior_box_size) +
+                                          "). Current value is: ",
+                                      proposals_pshape[2].get_length(),
+                                      ".");
+            }
+        }
+    }
+
+    if (get_input_size() > 3)
     {
-        auto box_logits_shape = get_input_partial_shape(0).to_shape();
-        set_output_type(
-            0, element::f32, Shape{1, 1, m_attrs.keep_top_k[0] * box_logits_shape[0], 7});
+        const PartialShape& aux_class_preds_pshape = get_input_partial_shape(3);
+        const PartialShape& aux_box_preds_pshape = get_input_partial_shape(4);
+        if (aux_class_preds_pshape.rank().is_static())
+        {
+            NODE_VALIDATION_CHECK(this,
+                                  aux_class_preds_pshape[0].compatible(num_images),
+                                  "Additional class predictions' first dimension must be "
+                                  "compatible with batch size.");
+            if (num_prior_boxes.is_static())
+            {
+                int num_prior_boxes_val = num_prior_boxes.get_length();
+                NODE_VALIDATION_CHECK(
+                    this,
+                    aux_class_preds_pshape[1].get_length() == num_prior_boxes_val * 2,
+                    "Additional class predictions' second dimension must be equal to "
+                    "num_prior_boxes * 2 (" +
+                        std::to_string(num_prior_boxes_val * 2) + "). Got " +
+                        std::to_string(aux_class_preds_pshape[1].get_length()) + ".");
+            }
+        }
+        NODE_VALIDATION_CHECK(
+            this,
+            aux_box_preds_pshape.compatible(box_logits_pshape),
+            "Additional box predictions' shape must be compatible with box logits shape.");
+    }
+
+    std::vector<Dimension> output_shape{1, 1};
+    if (num_images.is_dynamic())
+    {
+        output_shape.push_back(Dimension::dynamic());
     }
     else
     {
-        set_output_type(0, element::f32, PartialShape::dynamic());
+        size_t num_images_val = num_images.get_interval().get_min_val();
```

Comment:
Why do you use `num_images.get_interval().get_min_val()` here? Why don't use just `num_images`?

Reply:
right, I simplified that code a little.